### DISCUSSION TITLE
怎么为不同的target分别指定debug和release？

### DISCUSSION THREAD

**Discussion by chen-qingyu** [xmake-io/xmake#4647] [Q&A] [answered]
比如我有两个target，怎么简单的设置一个以debug模式编译运行，另一个以release模式编译运行？我知道可以通过`xmake f -m debug`这样来切换，但有没有什么简单的办法使得`xmake build`就全部搞定？（一个是unit test需要debug模式编译运行，一个是benchmark需要release模式编译运行）

```cpp
add_rules("mode.debug", "mode.release")
set_languages("cxx17")

target("foo") -- using debug
    set_kind("binary")
    add_files("foo/*.cpp")

target("bar") -- using release
    set_kind("binary")
    add_files("bar/*.cpp")
```

**Comment by waruqi** [comment, marked as answer]
不能，你可以定义自己的 mode ，分别对 两个 target 设置不同的配置，一个用 debug 的配置，一个用 release 的配置。。
